jubilant/_juju.py

```diff
@@ -205,6 +205,151 @@ def add_unit(
+                credentials. See ``regions`` for a list of available regions for a given cloud.
+            add_model: Name of an initial model to create on the new controller.
+            agent_version: Version of agent binaries to use for Juju agents.
+            auto_upgrade: If true, after bootstrap, upgrade to the latest patch release.
+            bootstrap_base: Specify the base of the bootstrap machine.
+            bootstrap_constraints: Specify bootstrap machine constraints, for example,
+                ``{'mem': '8G'}``. If used, its values will also apply to any future
+                controllers provisioned for high availability (HA).
+            bootstrap_image: Specify the image of the bootstrap machine (requires
+                bootstrap_constraints specifying architecture).
+            bootstrap_series: Specify the series of the bootstrap machine (deprecated, use
+                bootstrap_base).
+            build_agent: If true, build local version of agent binary before bootstrapping.
+            clouds: If true, print the available clouds which can be used to bootstrap a
+                Juju environment.
+            config: Controller configuration options. Model config keys only affect the
+                controller model.
+            constraints: Set model constraints, for example, ``{'mem': '8G', 'cores': '4'}``.
+                If used, its values will be set as the default constraints for all future
+                workload machines in the model, exactly as if the constraints were set with
+                ``juju set-model-constraints``.
+            controller_charm_channel: The Charmhub channel to download the controller charm
+                from (if not using a local charm).
+            controller_charm_path: Path to a locally built controller charm.
+            credential: Credentials to use when bootstrapping.
+            db_snap: Path to a locally built .snap to use as the internal juju-db service.
+            db_snap_asserts: Path to a local .assert file (requires ``db_snap``).
+            force: If true, allow bypassing of checks such as supported series.
+            keep_broken: If true, do not destroy the provisioned controller instance if
+                bootstrap fails.
+            metadata_source: Local path to use as agent and/or image metadata source.
+            model_default: Configuration options to set for all models, unless otherwise
+                specified.
+            no_switch: If true, do not switch to the newly created controller.
+            regions: Print the available regions for the specified cloud.
+            storage_pool: Options for an initial storage pool as key-value pairs. ``name``
+                and ``type`` are required, plus any additional attributes.
+            to: Placement directive indicating an instance to bootstrap.
+
+        Returns:
+            The standard output of the Juju CLI command.
```

Comment:
Note that the standard output of `juju bootstrap` is empty. The stderr contains the interesting stuff (I just checked with `juju bootstrap lxd lxd2 >out 2>err`). So if anything we should return stderr.

However, I think maybe we should just return nothing, as it's human-readable and has to be parsed anything, so is not particularly useful. It may be useful for diagnostics, though, so how about we log it and return nothing?

```python
_, stderr = self._cli(*args, include_model=False)
logger.info('bootstrap output:\n%s', stderr)
```